packages/devtools_app/lib/src/screens/memory/panes/allocation_tracing/allocation_profile_tracing_tree.dart

```diff
@@ -0,0 +1,274 @@
+          );
+        } else if (selection.traceAllocations &&
+            controller.selectedTracedClassAllocationData == null) {
+          errorColumn = Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              Text(
+                'No allocation samples have been collected for class ${selection.cls.name}.\n',
+              ),
+            ],
+          );
+        }
+        if (errorColumn != null) {
+          return errorColumn;
+        }
+        late Function(Function()) updateTreeStateCallback;
+        final theme = Theme.of(context);
+        return Column(
+          children: [
+            AreaPaneHeader(
+              title: Text.rich(
+                TextSpan(
+                  children: [
+                    const TextSpan(
+                      text: 'Traced allocations for: ',
+                    ),
+                    TextSpan(
+                      style: theme.fixedFontStyle,
+                      text: controller.selectedTracedClass.value?.cls.name!,
+                    ),
+                  ],
+                ),
+              ),
+              tall: true,
+              needsTopBorder: false,
+              actions: [
+                ExpandAllButton(
+                  onPressed: () => updateTreeStateCallback(
+                    () {
+                      final data = controller.selectedTracedClassAllocationData;
+                      if (data == null) {
+                        return;
+                      }
+                      for (final root in data.bottomUpRoots) {
+                        root.expandCascading();
+                      }
+                    },
+                  ),
+                ),
+                const SizedBox(width: denseSpacing),
+                CollapseAllButton(
+                  onPressed: () => updateTreeStateCallback(
+                    () {
+                      final data = controller.selectedTracedClassAllocationData;
+                      if (data == null) {
+                        return;
+                      }
+                      for (final root in data.bottomUpRoots) {
+                        root.collapseCascading();
+                      }
+                    },
+                  ),
+                ),
+              ],
+            ),
+            Expanded(
+              child: StatefulBuilder(
+                builder: (context, setState) {
+                  updateTreeStateCallback = setState;
+                  return _AllocationProfileTracingCallTreeTable(
+                    cls: selection!.cls,
+                    // TODO(bkonyi): support call stack and bottom up views.
+                    dataRoots: controller
+                        .selectedTracedClassAllocationData!.bottomUpRoots,
+                  );
+                },
+              ),
+            ),
+          ],
+        );
+      },
+    );
+  }
+
+  Widget _allocationTracingInstructions(BuildContext context) {
```

Comment:
make this a stateless widget class instead of a private method that returns a widget